tools/shared.py

```diff
@@ -1934,12 +1955,14 @@ def link_lld(args, target, opts=[], lto_level=0):
     if Settings.EXPORT_ALL:
       cmd += ['--export-all']
 
-    logger.debug('emcc: lld-linking: %s to %s', args, target)
-    check_call(cmd)
+    cmd += opts
+
+    run_process(cmd)
```

Comment:
AFAICT this will change from existing on failure, to raise and exception and showing a backtrace.  Seems like we want check_call here.

Reply:
`run_process` checks the output by default, so it will throw an error - isn't the error similar to what `check_call` does?

Reply:
(`check_call` does throw an error with a backtrace as well, I verified)

Reply:
Really, check_call looks like it should always tries to exit cleanly (not backtrace).

We want to see a backtrace when an subtool fails (e.g. when the linker fails.) due to bad user input.    At least that has been my goal for a while now.  Users should only see a backtrace if something went wrong with emscripten itself, not their input.

Reply:
Oh, right, I was mixing up python's `check_call` and ours (which maybe suggests we should rename ours?) Will fix.